hibernate-reactive-core/src/main/java/org/hibernate/reactive/session/impl/ReactiveStatelessSessionImpl.java

```diff
@@ -0,0 +1,418 @@
+/* Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ * Copyright: Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.reactive.session.impl;
+
+import org.hibernate.HibernateException;
+import org.hibernate.LockMode;
+import org.hibernate.LockOptions;
+import org.hibernate.UnresolvableObjectException;
+import org.hibernate.cache.spi.access.EntityDataAccess;
+import org.hibernate.engine.internal.Versioning;
+import org.hibernate.engine.query.spi.HQLQueryPlan;
+import org.hibernate.engine.query.spi.sql.NativeSQLQuerySpecification;
+import org.hibernate.engine.spi.PersistenceContext;
+import org.hibernate.engine.spi.QueryParameters;
+import org.hibernate.engine.spi.SharedSessionContractImplementor;
+import org.hibernate.id.IdentifierGenerationException;
+import org.hibernate.id.IdentifierGenerator;
+import org.hibernate.internal.SessionCreationOptions;
+import org.hibernate.internal.SessionFactoryImpl;
+import org.hibernate.internal.StatelessSessionImpl;
+import org.hibernate.jpa.spi.NativeQueryTupleTransformer;
+import org.hibernate.loader.custom.CustomQuery;
+import org.hibernate.persister.entity.EntityPersister;
+import org.hibernate.query.ParameterMetadata;
+import org.hibernate.query.Query;
+import org.hibernate.query.internal.ParameterMetadataImpl;
+import org.hibernate.reactive.common.ResultSetMapping;
+import org.hibernate.reactive.engine.impl.ReactivePersistenceContextAdapter;
+import org.hibernate.reactive.id.ReactiveIdentifierGenerator;
+import org.hibernate.reactive.loader.custom.impl.ReactiveCustomLoader;
+import org.hibernate.reactive.persister.entity.impl.ReactiveEntityPersister;
+import org.hibernate.reactive.pool.ReactiveConnection;
+import org.hibernate.reactive.session.ReactiveNativeQuery;
+import org.hibernate.reactive.session.ReactiveQuery;
+import org.hibernate.reactive.session.ReactiveStatelessSession;
+import org.hibernate.reactive.util.impl.CompletionStages;
+import org.hibernate.type.IntegerType;
+import org.hibernate.type.LongType;
+import org.hibernate.type.Type;
+
+import javax.persistence.Tuple;
+import java.io.Serializable;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * An {@link ReactiveStatelessSession} implemented by extension of
+ * the {@link StatelessSessionImpl} in Hibernate core. Extension was
+ * preferred to delegation because there are places where
+ * Hibernate core compares the identity of session instances.
```

Comment:
I don't understand the concern regarding identity of session instances. Surely you don't expect people to use both reactive and non-reactive sessions within the same SessionFactory instance?

Reply:
No, it's nothing to do with that.

And actually TBH I'm not sure if this comment really applies to `StatelessSession`s (though it might). It's a copy/paste from the comment in `ReactiveSessionImpl`. The issue is that Hibernate core sometimes does `if (session==otherSession)` which caused problems when we had  `ReactiveSessionImpl` delegating to `SessionImpl`.

Here I've just followed the design that worked for `ReactiveSessionImpl`.

Reply:
ok, feel free to keep it. Maybe adjust the comment if you're not sure?